auth/src/main/java/com/firebase/ui/auth/ui/phone/CheckPhoneNumberFragment.java

```diff
@@ -0,0 +1,197 @@
+                mPhoneInputLayout.setError(null);
+            }
+        });
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        if (savedInstanceState != null || mCalled) { return; }
+        // Fragment back stacks are the stuff of nightmares (what's new?): the fragment isn't
+        // destroyed so its state isn't saved and we have to rely on an instance field. Sigh.
+        mCalled = true;
+
+        // Check for phone
+        // It is assumed that the phone number that are being wired in via Credential Selector
+        // are e164 since we store it.
+        Bundle params = getArguments().getBundle(ExtraConstants.PARAMS);
+        String phone = null;
+        String countryIso = null;
+        String nationalNumber = null;
+        if (params != null) {
+            phone = params.getString(ExtraConstants.PHONE);
+            countryIso = params.getString(ExtraConstants.COUNTRY_ISO);
+            nationalNumber = params.getString(ExtraConstants.NATIONAL_NUMBER);
+        }
+
+        if (!TextUtils.isEmpty(countryIso) && !TextUtils.isEmpty(nationalNumber)) {
+            start(PhoneNumberUtils.getPhoneNumber(countryIso, nationalNumber));
+        } else if (!TextUtils.isEmpty(countryIso)) {
+            setCountryCode(new PhoneNumber(
+                    "",
+                    countryIso,
+                    String.valueOf(PhoneNumberUtils.getCountryCode(countryIso))));
+        } else if (!TextUtils.isEmpty(phone)) {
+            start(PhoneNumberUtils.getPhoneNumber(phone));
+        } else if (getFlowParams().enableHints) {
+            FlowUtils.unhandled(this, new PendingIntentRequiredException(
+                    Credentials.getClient(mHandler.getApplication()).getHintPickerIntent(
+                            new HintRequest.Builder().setPhoneNumberIdentifierSupported(true)
+                                    .build()),
+                    RequestCodes.CRED_HINT));
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode != RequestCodes.CRED_HINT || resultCode != Activity.RESULT_OK) { return; }
+
+        Credential credential = data.getParcelableExtra(Credential.EXTRA_KEY);
+        String formattedPhone = PhoneNumberUtils.formatUsingCurrentCountry(
+                credential.getId(), getContext());
+        if (formattedPhone != null) {
+            start(PhoneNumberUtils.getPhoneNumber(formattedPhone));
+        }
+    }
+
+    @Override
+    public void onClick(View v) {
+        onNext();
+    }
+
+    private void start(PhoneNumber number) {
+        if (PhoneNumber.isValid(number)) {
+            mPhoneEditText.setText(number.getPhoneNumber());
+            mPhoneEditText.setSelection(number.getPhoneNumber().length());
+        }
+        if (PhoneNumber.isCountryValid(number)) {
+            setCountryCode(number);
```

Comment:
If `isCountryValid` is `true` but `isValid` is `false` then it looks like we'll call `onNext()` with a situation that will cause `getPseudoValidPhoneNumber()` to return null and then put an error on screen.

Reply:
Oops, you're right.